keras_nlp/tokenizers/sentence_piece_tokenizer_trainer_test.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for SentencePiece Tokenizer Trainer."""
+
+import os
+
+import tensorflow as tf
+
+from keras_nlp.tokenizers.sentence_piece_tokenizer import SentencePieceTokenizer
+from keras_nlp.tokenizers.sentence_piece_tokenizer_trainer import (
+    compute_sentencepiece_vocabulary,
+)
+
+
+class SentencePieceTokenizerTrainerTest(tf.test.TestCase):
+    def test_dataset_input(self):
+        test_text = ["Ninjas and Samurais"]
+        expected_output = [
+            [5, 9, 6, 7, 11, 4, 8, 5, 4, 7, 10, 5, 12, 4, 13, 15, 14, 4, 6, 8]
+        ]
+        data = tf.data.Dataset.from_tensor_slices(test_text)
+        proto = compute_sentencepiece_vocabulary(data, 16)
+        tokenizer = SentencePieceTokenizer(proto=proto)
+        test_output = tokenizer(test_text)
+        self.assertAllEqual(expected_output, test_output)
+
+    def test_file_input(self):
+        test_text = "Ninja Land"
+        with open("test.txt", "w+") as f:
```

Comment:
I think you should make sure to open all these files inside self.get_temp_dir()

https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/tokenizers/sentence_piece_tokenizer_test.py#L188

Make sure there's no collisions for different tests reading writing the same files, and that things get cleaned up.